src/service/video-manager-impl.js

```diff
@@ -284,43 +299,111 @@ class VideoEntry {
+ * Note that even if platfrom supports autoplay, users or browsers can disable
+ * autoplay to save data / battery. This function detects both platfrom support
+ * and when autoplay is disabled.
+ *
+ * Service dependencies are taken explicitly for testability.
+ *
+ * @private visible for testing.
+ * @param {!./ampdoc-impl.AmpDoc} ampdoc
+ * @param {!./platform-impl.Platform} platform
+ * @param {!./timer-impl.Timer} timer
+ * @param {!../mode.ModeDef} mode
+ * @return {!Promise<boolean>}
  */
-function platformSupportsAutoplay(win) {
-  // Do not support autoplay in amp-lite viewer
-  if (getMode(win).lite) {
-    return false;
+export function supportsAutoplay(ampdoc, platform, timer, mode) {
+
+  // Use cached result if available
+  if (supportsAutoplayCache_) {
+    return supportsAutoplayCache_;
   }
 
-  const platform = platformFor(win);
-  // non-mobile platforms always supported autoplay
-  if (!platform.isAndroid() && !platform.isIos()) {
-    return true;
+  // We do not support autoplay in amp-lite viewer regardless of platfrom
+  if (mode.lite) {
+    return supportsAutoplayCache_ = Promise.resolve(false);
   }
 
+  // Short-circuit for known unsupported versions on mobile
   const version = platform.getMajorVersion();
-  if (platform.isAndroid()) {
-    if (platform.isFirefox() && version >= 37) {
-      return true;
-    }
-    if (platform.isChrome() && version >= 53) {
-      return true;
-    }
+  if (platform.isChrome() && version < 53) {
+    return supportsAutoplayCache_ = Promise.resolve(false);
   }
 
-  if (platform.isIos()) {
-    if (platform.isSafari() && version >= 10) {
-      return true;
-    }
+  if (platform.isSafari() && version < 10) {
+    return supportsAutoplayCache_ = Promise.resolve(false);
   }
 
-  // TODO(aghassemi): Test other combinations and add support.
-  // TODO(aghassemi): Is there a way to detect that autoplay has been disabled
-  // in the user preferences of the browser?
-  return false;
+  // To detect autoplay, we create a video element with a inline data URI
+  // and call `play()` on it.
+  const detectionElement = ampdoc.win.document.createElement('video');
+  detectionElement.setAttribute('muted', '');
+  detectionElement.setAttribute('playsinline', '');
+  detectionElement.setAttribute('webkit-playsinline', '');
+  detectionElement.setAttribute('height', '0');
+  detectionElement.setAttribute('width', '0');
+  setStyles(detectionElement, {
+    position: 'absolute',
+    top: '0',
+    width: '0px',
+    height: '0px',
+    opacity: '0',
+  });
+
+  // The following video file is an h.264 encoded two-second video with four
+  // black frames generated with wide decoding compatibility in mind.
+  // File is 909 bytes (1212 in base64).
+  // Video file is generated by the following `ffmpeg` command:
+  //  $ffmpeg -f lavfi -i color=color=black:rate=2:size=100x100 -t 2
+  //  -profile:v baseline -preset slow
+  //  -pix_fmt yuv420p -vcodec libx264 /tmp/small.mp4
+  /*eslint-disable */
```

Comment:
1212 bytes base64 encoded as a utf8 string. (raw video is 909 bytes). Certainly possible to host it on `cdn.ampproject.org` that was my first plan when I thought I could not get the video less than 3k. 

I think if we manage not to make this a remote dependency, it would be better. Otherwise we need to increate the timeout of playing promise to account for connectivity issues and it could give us false negatives in some situations. 

As per your next comment, I do think we should make <amp-video> its own extension (soon) and move this service out of core, installed only by video players that implement the interface. video-manager is certainly going to grow in size as we add more code for other features.